tests/cases/fourslash/codeFixInferFromUsageRestParam2JS.ts

```diff
@@ -0,0 +1,27 @@
+/// <reference path='fourslash.ts' />
+
+// @allowJs: true
+// @checkJs: true
+// @noImplicitAny: true
+// @Filename: important.js
+/////** @param {number} a */
+////function f(a, [|...rest|]){
+////    a; rest;
+////}
+////f(1);
+////f(2, "s1");
+////f(3, false, "s2");
+////f(4, "s1", "s2", false, "s4");
+
+verify.fileAfterCodeFix(
+`/** @param {number} a */
+/**

```

Comment:
Could you file this as a bug? We should combine comments, not add a new one.

Reply:
I'll file it after merging the PR since the bug depends on being able to emit JSDoc in the first place.

I personally think the current approach is a good compromise and that we'll rarely have to generate partial JSDoc. The bigger lack is untyped JSDoc:

```js
/** @param x Just a description, no type */
function f(x) {
    return x * x
}
```